shell/shared/grain-shared.js

```diff
@@ -57,6 +57,21 @@ Meteor.methods({
                         trashed: { $exists: false }, },
                        { $set: { "trashed": new Date() } },
                        { multi: true });
+
+      const grainsOwned = Grains.find({
+        userId: { $eq: this.userId },
+        _id: { $in: grainIds },
+      }, { fields: { _id: 1, }, });
+      grainsOwned.forEach((grain) => {
+        try {
+          Sessions.remove({ grainId: grain._id, });
```

Comment:
Seems like this shouldn't be in the try/catch? Then the entire try/catch can be in if (!this.isSimulation).